src/Layer.js

```diff
@@ -66,18 +69,24 @@ export default class Layer {
         this._emitter = mitt();
         this._renderLayer = new RenderLayer();
 
+        this._initializeUIDS();
+        this._sourcePromise = this.update(source, viz);
+        this._renderWaiters = [];
+    }
+
+    /**
+     * Initialize unique identifiers to detect concurrent updates
+     */
+    _initializeUIDS () {
```

Comment:
I'm not a big fan of moving _initializers_ outside of the constructor.

Reply:
I'm not either. I had found the constructor a bit bloated and I was trying to reduce somehow its responsibilities. 

But your comment made me think about it and now I think I can refactor it a bit, extracting the UIDs mechanism. 👍

Reply:
@rochoa check it please when you have a pair of minutes... 👇 commits

Reply:
@rochoa you can check it if you want... (👇 commits). I'll merge this on Monday

Reply:
Much better, isn't it? 😄